contracts/src/starknet/token_bridge.cairo

```diff
@@ -152,28 +159,29 @@ func initiate_withdraw{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_c
     assert message_payload[2] = l1_recipient
     assert message_payload[3] = amount.low
     assert message_payload[4] = amount.high
-    
-    send_message_to_l1(to_address=to_address, payload_size=5, payload=message_payload)
 
+    send_message_to_l1(to_address=to_address, payload_size=5, payload=message_payload)
+    withdraw_initiated.emit(l1_recipient, amount, caller_address)
     return ()
 end
 
 @l1_handler
 func handle_deposit{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
-        from_address : felt, l2_recipient : felt, l2_token_address: felt, amount_low: felt, amount_high: felt):
+        from_address : felt, l2_recipient : felt, l2_token_address : felt, amount_low : felt,
+        amount_high : felt):
     # The amount is validated (i.e. amount_low, amount_high < 2**128) by an inner call to
     # IMintableToken permissionedMint function.
 
     let (expected_from_address) = get_l1_token_bridge()
     with_attr error_message("Expected deposit from l1_token_bridge"):
         assert from_address = expected_from_address
     end
-    let amount : Uint256 = cast((low=amount_low, high=amount_high), Uint256)
+    let amount = Uint256(low=amount_low, high=amount_high)
 
     assert_not_zero(l2_token_address)
 
     # Call mint on l2_token contract.
     IL2Token.mint(contract_address=l2_token_address, recipient=l2_recipient, amount=amount)
-
+    deposit_handled.emit(l2_recipient, amount)
```

Comment:
I think we should also emit which token it was

Reply:
I agree